src/WebJobs.Extensions.DurableTask/Listener/DurableTaskScaleMonitor.cs

```diff
@@ -0,0 +1,180 @@
+                if (heartbeat.WorkItemQueueLatency != null)
+                {
+                    metrics.WorkItemQueueLatency = heartbeat.WorkItemQueueLatency.ToString();
+                }
+            }
+
+            return metrics;
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.Cast<DurableTaskTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<DurableTaskTriggerMetrics> context)
+        {
+            return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusCore(int workerCount, DurableTaskTriggerMetrics[] metrics)
+        {
+            var scaleStatus = new ScaleStatus() { Vote = ScaleVote.None };
+            if (metrics == null)
+            {
+                return scaleStatus;
+            }
+
+            var heartbeats = new PerformanceHeartbeat[metrics.Length];
+            for (int i = 0; i < metrics.Length; ++i)
+            {
+                TimeSpan workItemQueueLatency;
+                bool parseResult = TimeSpan.TryParse(metrics[i].WorkItemQueueLatency, out workItemQueueLatency);
+
+                heartbeats[i] = new PerformanceHeartbeat
+                {
+                    PartitionCount = metrics[i].PartitionCount,
+                    ControlQueueLengths = JsonConvert.DeserializeObject<IReadOnlyList<int>>(metrics[i].ControlQueueLengths),
+                    ControlQueueLatencies = JsonConvert.DeserializeObject<IReadOnlyList<TimeSpan>>(metrics[i].ControlQueueLatencies),
+                    WorkItemQueueLatency = parseResult ? workItemQueueLatency : TimeSpan.FromMilliseconds(0),
+                    WorkItemQueueLength = metrics[i].WorkItemQueueLength,
+                };
+            }
+
+            var scaleRecommendation = this.PerformanceMonitor.MakeScaleRecommendation(workerCount, heartbeats.ToArray());
+
+            switch (scaleRecommendation?.Action)
+            {
+                case ScaleAction.AddWorker:
+                    scaleStatus.Vote = ScaleVote.ScaleOut;
+                    break;
+                case ScaleAction.RemoveWorker:
+                    scaleStatus.Vote = ScaleVote.ScaleIn;
+                    break;
+                default:
+                    scaleStatus.Vote = ScaleVote.None;
+                    break;
+            }
+
+            this.traceHelper.ExtensionInformationalEvent(
+                                    this.hubName,
+                                    string.Empty,
+                                    this.functionName.Name,
+                                    $"Durable Functions Trigger Scale Decision: {scaleStatus.Vote.ToString()}, Reason: {scaleRecommendation?.Reason}",
+                                    writeToUserLogs: false);

```

Comment:
Any particular reason for these to *not* go in user logs? Is there a middle ground where we write to user logs for just `ScaleVote.ScaleOut` and `ScaleVote.ScaleIn`?

Reply:
No, no strong reason. I can make that change.